scene/3d/ray_cast.cpp

```diff
@@ -231,6 +240,47 @@ void RayCast::_bind_methods() {
 	ADD_PROPERTY(PropertyInfo(Variant::INT, "type_mask", PROPERTY_HINT_FLAGS, "Static,Kinematic,Rigid,Character,Area"), "set_type_mask", "get_type_mask");
 }
 
+void RayCast::_create_debug_shape() {
+
+	if (debug_shape) {
+		debug_shape->queue_delete();
+		debug_shape = NULL;
+	}
+
+	Vector<Vector3> vertices;
+	vertices.push_back(Vector3());
+	vertices.push_back(cast_to);
+
+	Ref<Mesh> mesh = memnew(Mesh);
+	Array a;
+	a.resize(Mesh::ARRAY_MAX);
+
+	a[Mesh::ARRAY_VERTEX] = vertices;
+
+	PoolVector<Color> color;
+	color.resize(vertices.size());
+	{
+		PoolVector<Color>::Write w = color.write();
+		for (int i = 0; i < vertices.size(); i++) {
+			if (collided)
+				w[i] = Color(0, 0, 0, 1);
```

Comment:
No observable change in color in spite of this. I've confirmed with gdb that we are in fact hitting this case.